Games/types/Mafia/roles/cards/NaughtyOrNice.js

```diff
@@ -0,0 +1,46 @@
+const Card = require("../../Card");
+const { LABEL_INVESTIGATE, LABEL_ALIGNMENT } = require("../../const/ActionLabel");
+const { FLAG_VOTING } = require("../../const/MeetingFlag");
+const { PRIORITY_ALIGNMENT_LEARNER } = require("../../const/Priority");
+const { STATE_NIGHT } = require("../../const/States");
+
+module.exports = class NaughtyOrNice extends Card {
+
+    constructor(role) {
+        super(role);
+
+        this.meetings = {
+            "Check List": {
+                states: [ STATE_NIGHT ],
+                flags: [ FLAG_VOTING ],
+                action: {
+                    labels: [ LABEL_INVESTIGATE, LABEL_ALIGNMENT ],
+                    priority: PRIORITY_ALIGNMENT_LEARNER,
+                    run: function () {
+                        if (!this.actor.role.data.visitors?.length)
```

Comment:
a small point: a guard clause can help reduce the nesting here

if (this.actor.role.data.visitors?.length) { return }